core/src/main/java/io/questdb/griffin/engine/functions/finance/RegressionSlopeFunctionFactory.java

```diff
@@ -0,0 +1,190 @@
+                aggregate(mapValue, x, y);
+            }
+        }
+
+        @Override
+        public int getValueIndex() {
+            return valueIndex;
+        }
+
+        @Override
+        public void initValueIndex(int valueIndex) {
+            this.valueIndex = valueIndex;
+        }
+
+        @Override
+        public void initValueTypes(ArrayColumnTypes columnTypes) {
+            this.valueIndex = columnTypes.getColumnCount();
+            columnTypes.add(ColumnType.DOUBLE);
+            columnTypes.add(ColumnType.DOUBLE);
+            columnTypes.add(ColumnType.DOUBLE);
+            columnTypes.add(ColumnType.DOUBLE);
+            columnTypes.add(ColumnType.LONG);
+        }
+
+        @Override
+        public void setNull(MapValue mapValue) {
+            mapValue.putDouble(valueIndex, Double.NaN);
+            mapValue.putDouble(valueIndex + 1, Double.NaN);
+            mapValue.putDouble(valueIndex + 2, Double.NaN);
+            mapValue.putDouble(valueIndex + 3, Double.NaN);
+            mapValue.putLong(valueIndex + 4, 0);
+        }
+
+        @Override
+        public boolean supportsParallelism() {
+            return false;
+        }
+
+
+        @Override
+        public double getDouble(Record rec) {
+            long count = rec.getLong(valueIndex + 4);
+
+            if (count == 0) {
+                return Double.NaN;
+            }
+
+            double covXY = rec.getDouble(valueIndex + 3) / count;
+            double varX = rec.getDouble(valueIndex + 2) / count;
+
+            if (varX == 0) {
+                return Double.NaN;
+            }
+
+            return covXY / varX;
+        }
+
+        @Override
+        public Function getLeft() {
+            return xFunction;
+        }
+
+        @Override
+        public Function getRight() {
+            return yFunction;
+        }
+
+        @Override
+        public boolean isConstant() {
+            return false;
+        }
+
+        @Override
+        public String getName() {
+            return "regr_slope";
+        }
+
+        protected void aggregate(MapValue mapValue, double x, double y) {
+            double meanX = mapValue.getDouble(valueIndex);
+            double meanY = mapValue.getDouble(valueIndex + 1);
+            double sumX = mapValue.getDouble(valueIndex + 2);
+            double sumXY = mapValue.getDouble(valueIndex + 3);
+            long count = mapValue.getLong(valueIndex + 4) + 1;
+
+            double oldMeanX = meanX;
+            meanX += (x - meanX) / count;
+            meanY += (y - meanY) / count;
+            sumX += (x - oldMeanX) * (x - meanX);
+            sumXY += (x - oldMeanX) * (y - meanY);
+
+            mapValue.putDouble(valueIndex, meanX);
+            mapValue.putDouble(valueIndex + 1, meanY);
+            mapValue.putDouble(valueIndex + 2, sumX);
+            mapValue.putDouble(valueIndex + 3, sumXY);
+            mapValue.addLong(valueIndex + 4, 1L);
```

Comment:
nit: the `count` is already calculated above, this could just be:

```java
mapValue.put(valueIndex + 4, count);
```

`addValue()` will do another non-optimised memory read unnecessarily.